### PR TITLE
Feat/firebase auth

### PR DESCRIPTION
**Pull request by rcaina**
Add Authentication for app using Firebase Authentication! (No need for Amplify or Cognito. Authentication is through email, SMS will be added later on)

- New packages: 
 -- firebase/auth: Allows Authentication through Firebase
 
- Main Updated Files: Login, SignUp, Confirmation, AuthService, AuthWrapper
- Removed: Amplify, Encrypted-storage (Firebase takes care of persistent login), unused feature "Demo snitch" from settings.  

One thing to note, testing will be important. At times when I logged in (using emulator), it would take quite awhile for the verification email to be sent. I would have to wait until it was finally triggered and the email was sent. So if you do not get an email right away when testing, just wait awhile. Not sure if this is just because of the emulator or connection or what. But in the end it does work.

### CODE REVIEW

File: react-native-app/services/AuthService.ts [hunk 1]
```diff
@@ -1,53 +1,74 @@
...
+	async attemptLogin(email:string, password:string): Promise<User | undefined> {
+		return await auth()
+			.signInWithEmailAndPassword(email, password)
+			.then(async () => {
+				const input:any = auth().currentUser;
+				console.log('User account created & signed in!');
+				return await ServerFacade.getUserById(input.uid);
+			})
+			.catch(error => {
+				if (error.code === 'auth/email-already-in-use') {
```
Comment (amj311) [inline react-native-app/services/AuthService.ts]: Is 'auth/email-already-in-use' an error we would expect to get back from a login attempt?

Comment (amj311) [reply react-native-app/services/AuthService.ts]: Are there error options here for an invalid password?

File: react-native-app/services/AuthService.ts [hunk 2]
```diff
@@ -1,53 +1,74 @@
...
+				console.error(error);
+				return undefined;
+			});
 	},
 
+	async resendConfirmationEmail()  {
+		try{
+			await auth().currentUser?.sendEmailVerification();
+		} catch(error){
+			console.log({error})
```
Comment (amj311) [inline react-native-app/services/AuthService.ts]: We should probably display a message to the user if there is an error with the verification email. We wouldn't want to make them wait on nothing

Comment (amj311) [inline react-native-app/services/AuthService.ts]: We should probably display a message to the user if there is an error with the verification email. We wouldn't want to make them wait on nothing

**Review by amj311 — CHANGES_REQUESTED** [review]
This looks great! So much simpler. Just a few tiny comments about some error catching

File: react-native-app/services/AuthService.ts (hunk 1 above)
Comment (rcaina) [reply react-native-app/services/AuthService.ts]: These were built in errors that came as part of the code snippet directly from firebase. I could look into invalid password error options.

File: react-native-app/services/AuthService.ts (hunk 2 above)
Comment (rcaina) [reply react-native-app/services/AuthService.ts]: I think i completely forgot about the error handling when putting this together haha. I was so focused on getting it all to pass.

Comment (rcaina) [reply react-native-app/services/AuthService.ts]: will do

**Review by amj311 — APPROVED** [review]

**Pull request merged**